cmd/bridge/main.go

```diff
@@ -232,6 +249,7 @@ func main() {
 		ThanosPublicURL:       thanosPublicURL,
 		LoadTestFactor:        *fLoadTestFactor,
 		InactivityTimeout:     *fInactivityTimeout,
+		DevCatalogCategories:  fs.Lookup("developer-catalog-categories").Value.String(),
```

Comment:
We need to be doing some validation here. We should fail if the JSON is not well-formed or the content doesn't match what we expect.

Reply:
Added validation for the dev catalog categories, see new `pkg/serverconfig/validate.go` and `pkg/serverconfig/validate_test.go`. It fails when required fields "id" or "label" is missing. When using the cli argument is validates also that the input could parsed as json, is an array, doesn't contain invalid keys.

When using a yaml file, it was parsed by `config.yaml`. In this case it only validates that there is an id and label prop.

This works:

```yaml
apiVersion: console.openshift.io/v1
kind: ConsoleConfig
customization:
  developerCatalog:
    categories: []
```

This crashs:

```yaml
apiVersion: console.openshift.io/v1
kind: ConsoleConfig
servingInfo:
  bindAddress: http://localhost:9000
customization:
  developerCatalog:
    categories:
      -
```

Output:

```bash
go run github.com/openshift/console/cmd/bridge -config bind-address-config.yaml
Developer catalog category at index 0 must have at least id and label properties.
exit status 1
```